### ISSUE TITLE
Cannot mix in non-composable type Any into object 

### ISSUE THREAD

**Issue by nkh** [rakudo/rakudo#2983]
on linux with rakudo:
This is Rakudo version 2019.03.1-616-g6565810f5 built on MoarVM version 2019.05-53-gcce71cdbf
implementing Perl 6.d.

from IRC: 
 <vrurg> nadim_: I don't know how the regex influences it, but the problem is that required_role is being exported as a class (i.e. it has ClassHOW meta).

The attached shell script generates files needed to reproduces the error and runs tests

[test_bug.sh.txt](https://github.com/rakudo/rakudo/files/3287667/test_bug.sh.txt)

**Comment by vrurg** [comment]
Brief note: the role somehow gets punned while imported and ends up as a `Metamodel::ClassHOW` instance.

**Comment by vrurg** [comment]
After having some fun fighting with bisectable, AlexDaniel managed to get the following [output from `commitable`](https://gist.github.com/Whateverable/fc766d48964f62a232d2bc00f30dd46c):

```
¦«5bbaabb9c6^^^^^^»:
Perl6::Metamodel::ParametricRoleHOW

¦«5bbaabb9c6^^^^^,5bbaabb9c6^»:
Cannot test this commit (Commit exists, but a perl6 executable could not be built for it)
¦«5bbaabb9c6^^^^,5bbaabb9c6^^^,5bbaabb9c6^^»:
Cannot find method 'FLATTENABLE_HASH' on object of type Mu
  in block <unit> at sandbox/role-import.pl line 4

 «exit code = 1»
¦«5bbaabb»:
Perl6::Metamodel::ClassHOW
```

**Comment by vrurg** [comment]
@nkh It is currently being decided what's to be done with this situation. The point is that `require` is not supposed to be working this way. If you check with documentation, it states that type object are not supposed to be imported this way. The mechanism was designed to import variables (sigilled entities).

**Comment by vrurg** [comment]
Closed via #3005
